docs/src/main/paradox/java/http/introduction.md

```diff
@@ -0,0 +1,139 @@
+
+Again the logic for the marshalling and unmarshalling JSON in this example is provided by the "spray-json" library
+(details on how to use that here: @ref[JSON Support](common/json-support.md#akka-http-spray-json))
+
+Read more about the details of the high level APIs in the section @ref[High-level Server-Side API](routing-dsl/index.md#http-high-level-server-side-api).
+
+## Low-level HTTP server APIs
+
+The low-level Akka HTTP server APIs allows for handling connections or individual requests by accepting
+`HttpRequest` s and answering them by producing `HttpResponse` s. This is provided by the `akka-http-core` module.
+APIs for handling such request-responses as function calls and as a `Flow<HttpRequest, HttpResponse, NotUsed>` are available.
+
+@@snip [HttpServerLowLevelExample.java](../../../../test/java/docs/http/javadsl/HttpServerLowLevelExample.java) { #low-level-server-example }
+
+Read more details about the low level APIs in the section @ref[Low-Level Server-Side API](server-side/low-level-server-side-api.md#http-low-level-server-side-api).
+
+## HTTP client API
+
+The client APIs provide methods for calling a HTTP server using the same `HttpRequest` and `HttpResponse` abstractions
+that Akka HTTP server uses but adds the concept of connection pools to allow multiple requests to the same server to be
+handled more performantly by re-using TCP connections to the server.
+
+Example simple request:
+
+@@snip [ClientSingleRequestExample.java](../../../../test/java/docs/http/javadsl/ClientSingleRequestExample.java) { #single-request-example }
+
+Read more about the details of the client APIs in the section @ref[Consuming HTTP-based Services (Client-Side)](client-side/index.md#http-client-side).
+
+## The modules that make up Akka HTTP
+
+Akka HTTP is structured into several modules:
+
+akka-http
+: Higher-level functionality, like (un)marshalling, (de)compression as well as a powerful DSL
+for defining HTTP-based APIs on the server-side, this is the recommended way to write HTTP servers
+with Akka HTTP. Details can be found in the section @ref[High-level Server-Side API](routing-dsl/index.md#http-high-level-server-side-api)
+
+akka-http-core
+: A complete, mostly low-level, server- and client-side implementation of HTTP (incl. WebSockets)
+Details can be found in sections @ref[Low-Level Server-Side API](server-side/low-level-server-side-api.md#http-low-level-server-side-api) and @ref[Consuming HTTP-based Services (Client-Side)](client-side/index.md#http-client-side)
+
+akka-http-testkit
+: A test harness and set of utilities for verifying server-side service implementations
+
+akka-http-spray-json
+: Predefined glue-code for (de)serializing custom types from/to JSON with [spray-json](https://github.com/spray/spray-json)
+Details can be found here: @ref[JSON Support](common/json-support.md#akka-http-spray-json)
```

Comment:
This should be removed - java does not care about spray :-)
See: See https://github.com/akka/akka-http/pull/521/files#r88125838

Toe example we want to point to here is jackson